### PR TITLE
Added description, list of algorithms and maintainers to readme

### PR DESCRIPTION
**Pull request by michalirak**
Response to Issue #14 
#### Added:
* [X]  List of algorithms that are currently implemented.
* [X]  A short description of the project.
* [X]  A list of maintainers of the project.

### CODE REVIEW

**Comment by NicholasACTran** [comment]
Can you add a note to encourage implementing already implemented algorithms in different languages?
Also, I don't think it's necessary to list every implementation in the README, but instead, just list the algorithms themselves

**Comment by yashasvi-goel** [comment]
@michalirak I second this(@NicholasACTran comment).

**Review by yashasvi-goel — COMMENTED** [review]
> Also, I don't think it's necessary to list every implementation in the README, but instead, just list the algorithms themselves

File: README.md [hunk 1]
```diff
@@ -21,6 +23,35 @@ The repo is structured like
     - Language
         - Implementation
 
+## Currently implemented algorithms
```
[the remark this answers is not in the document]
Comment (michalirak) [reply README.md:26]: I think we can request that if people add an algorithm they should add it to the README themselves.
If we go according to [this comment](https://github.com/yashasvi-goel/Algorithms-Open-Source/pull/33#issuecomment-703005648) then there is also less to update - only when a new algorithm is added, not when it is implemented in a new language.

**Comment by michalirak** [comment]
> Also, I don't think it's necessary to list every implementation in the README, but instead, just list the algorithms themselves

@NicholasACTran Shouldn't this mean that the repo should be structured accordingly? As in for each type of algorithm, a folder for that algorithm, and only inside it the implemented languages.

**Comment by NicholasACTran** [comment]
@michalirak What do you mean? Currently, it's structured Algorithm -> Language -> Implementation already. I was saying that the README should only have the Algorithms listed.

**Comment by michalirak** [comment]
@NicholasACTran Your suggestion means that if someone is searching for a specific algorithm, they will need to go through all the languages to find it's implementation. Since the readme only says which algorithms exist, it will not reflect the directory structure.
For example, if someone is looking for BFS traversal, they will see it exists, but to find it they will need to go through all the languages that exist in Graph Algorithms.
Either we include language in the readme which I agree is wasteful, or we should structure the repo so that the algorithm name is before the language, such as:
Algorithm Category -> Algorithm Type (if needed) -> Algorithm Name -> Language -> Implementation
Or, to narrow things down, without algorithm type.
I wished to comment this on your issue, but you seem to have closed it.

**Comment by NicholasACTran** [comment]
That level of restructuring is fine, but you would need to either open a new issue or restructure it in this PR. The other option is move the implementation details from a top-level README to subfolder READMEs.

**Comment by michalirak** [comment]
Fixed according to the comments. I think we should have a more structured discussion about the structure of this repo.

**Comment by michalirak** [comment]
> That level of restructuring is fine, but you would need to either open a new issue or restructure it in this PR. The other option is move the implementation details from a top-level README to subfolder READMEs.

I will open a new issue. I think too many READMEs will be more caotic than helpful, considering all these algorithms can be easily searched on google.

**Comment by NicholasACTran** [comment]
To be fair, implementations of all of these algorithms can also be easily searched on Google. I think the original idea was for the repo to be a one-stop shop for different algorithms. Discussion was happening somewhere else about having a README in the algorithm subfolders, giving a rundown about the theory of the algorithm and applications.

**Comment by yashasvi-goel** [comment]
I think having READMEs inside each folder for that folder would be better since we can have the reasons and explanations for each algorithm there.
[Check this]( https://github.com/yashasvi-goel/Algorithms-Open-Source/tree/master/Graph%20Algorithms/bi-dir-dijkstra )

**Comment by michalirak** [comment]
@yashasvi-goel Please merge soon. I fixed according to Issue #40 and the sooner this is merged the sooner we can make sure all contributions are done in the same way.

**Pull request merged**